vispy/app/backends/_qt.py

```diff
@@ -581,25 +596,33 @@ def _init_specific(self, p, kwargs):
                 raise RuntimeError('Cannot use vispy to share context and '
                                    'use built-in shareWidget.')
 
-        # first arg can be glformat, or a gl context
         if p.always_on_top or not p.decorate:
             hint = 0
             hint |= 0 if p.decorate else QtCore.Qt.FramelessWindowHint
             hint |= QtCore.Qt.WindowStaysOnTopHint if p.always_on_top else 0
         else:
             hint = QtCore.Qt.Widget  # can also be a window type
-        QGLWidget.__init__(self, glformat, p.parent, widget, hint)
+        if hasattr(QGLWidget, 'isSharing'):
+            # Qt4 and Qt5 < 5.4.0
+            QGLWidget.__init__(self, p.parent, widget, hint)
+        else:
+            # Qt5 >= 5.4.0
+            QGLWidget.__init__(self, p.parent, hint)
+        self.setFormat(glformat)
         self._initialized = True
-        if not self.isValid():
+        if hasattr(QGLWidget, 'isSharing') and not self.isValid():
             raise RuntimeError('context could not be created')
-        self.setAutoBufferSwap(False)  # to make consistent with other backends
+        if hasattr(self, 'setAutoBufferSwap'):
+            self.setAutoBufferSwap(False)  # to make consistent with other backends
```

Comment:
I can't find a way to disable automatic swapping; it's probably causing some of the troubles.